packages/shared.test/src/FauxFoundry/validateAction.ts

```diff
@@ -0,0 +1,331 @@
+          ...baseParam,
+        };
+        return;
+      }
+
+      for (const { name, fieldType, required } of dataType.fields) {
+        const fieldValue = (value as Record<string, unknown>)[name];
+        if (
+          (required && fieldValue == null)
+          || !matchesOntologyDataType(fieldType, fieldValue)
+        ) {
+          ret.result = "INVALID";
+          ret.parameters[paramKey] = {
+            ...baseParam,
+          };
+
+          return;
+        }
+      }
+
+      return;
+    }
+
+    case "objectType": {
+      if (
+        typeof value !== "string"
+        || !dataStore.ontology.getObjectTypeFullMetadata(value)
+      ) {
+        ret.result = "INVALID";
+        ret.parameters[paramKey] = {
+          ...baseParam,
+        };
+        return;
+      }
+
+      return;
+    }
+
+    default: {
+      const _assertNever: never = dataType;
+      throw new Error(
+        `validateDataType: unknown type`,
+      );
+    }
+  }
+}
+
+function matchesOntologyDataType(
+  odt: OntologyDataType,
+  value: unknown,
+): boolean {
+  switch (odt.type) {
+    case "any":
+      return true;
+    case "array":
+      return Array.isArray(value)
+        && value.every((v) => matchesOntologyDataType(odt.itemType, v));
+    case "binary":
+      throw new Error(`validateDataType: ${odt.type} not implemented yet.`);
+    case "boolean":
+      return typeof value === "boolean";
+    case "byte":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "cipherText":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "date":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "decimal":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "double":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "float":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "integer":
+      return (typeof value === "number" && Number.isInteger(value));
+
+    case "long":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "map":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "marking":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "object":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "objectSet":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
+      );
+    case "set":
+      throw new Error(
+        `matchesOntologyDataType: ${odt.type} not implemented yet.`,
```

Comment:
Left unimplemented things as their own block so that multiple people could work on this with fewer conflicts.